src/components/Preferences/CoursesPreferencesTable.tsx

```diff
@@ -0,0 +1,117 @@
+import { Select } from "@chakra-ui/react";
+import { useMemo, useState } from "react";
+import C2Table from "../C2Table";
+
+enum Willingness {
+    veryWilling = 2,
+    willing = 1,
+    unwilling = 0,
+}
+
+enum Difficulty {
+    difficult = 2,
+    moderate = 1,
+    easy = 0,
+}
+
+const CoursesPreferencesTable = (props) => {
+    const { courses } = props;
+
+    const columns = useMemo(
+        () => [
+            {
+                Header: "Course",
+                accessor: "course",
+            },
+            {
+                Header: "Willingness",
+                accessor: "willingness",
+                disableSortBy: true,
+                disableFilterBy: true,
+            },
+            {
+                Header: "Difficulty",
+                accessor: "difficulty",
+                disableSortBy: true,
+                disableFilterBy: true,
+            },
+        ],
+        []
+    );
+
+    // coursePreferences is dict with all the preferences
+    // {..., "CSC 225": {willingness: 1, difficulty: 0}}
+    const tableData = useMemo(() => makeTableData(courses), [courses]);
+
+    return <C2Table columns={columns} entries={tableData} />;
+};
+
+const makeTableData = (courses) => {
+    const [coursePreferences, setCoursePreferences] = useState(
```

Comment:
Can't use hook in a normal function - maybe remove for now seeing that the set function isn't being used?